ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/vlans.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.roles.eos_designs.python_modules.network_services.utils import UtilsMixin
+
+
+class VlansMixin(UtilsMixin):
+    """
+    Mixin Class used to generate structured config for one key.
+    Class should only be used as Mixin to a AvdStructuredConfig class
+    """
+
+    @cached_property
+    def vlans(self):
+        """
+        Return structured config for vlans
+
+        Consist of svis, mlag peering vlans and l2vlans from filtered tenants
+        """
+
+        if not self._network_services_l2:
+            return None
+
+        vlans = {}
+        for tenant in self._filtered_tenants:
+            for vrf in tenant["vrfs"]:
+                for svi in vrf["svis"]:
+                    vlan_id = int(svi["id"])
+                    vlans[vlan_id] = self._vlans_vlan(svi, tenant)
+
+                # MLAG IBGP Peering VLANs per VRF
+                # Continue to next VRF if mlag vlan_id is not set
+                if (vlan_id := self._mlag_ibgp_peering_vlan_vrf(vrf, tenant)) is None:
+                    continue
+
+                vlans[vlan_id] = {
+                    "tenant": tenant["name"],
+                    "name": f"MLAG_iBGP_{vrf['name']}",
+                    "trunk_groups": [self._trunk_groups_mlag_l3_name],
+                }
+
+            # L2 Vlans per Tenant
+            for l2vlan in tenant["l2vlans"]:
+                vlan_id = int(l2vlan["id"])
+                vlans[vlan_id] = self._vlans_vlan(l2vlan, tenant)
+
+        if vlans:
+            return vlans
+
+        return None
+
+    def _vlans_vlan(self, vlan, tenant):
```

Comment:
I would rename this `_get_vlan_config` or something like it - the name is a bit confusing

Reply:
fixed